lib/resque/worker.rb

```diff
@@ -861,7 +866,7 @@ def perform_with_fork(job, &block)
 
       begin
         @child = fork do
-          unregister_signal_handlers if term_child
+          unregister_signal_handlers
```

Comment:
This is as I explained in #1499. Below is what I wrote.

---------------------

However, in this way, the behavior of signal handler for a forked child must have been changed if TERM_CHILD environment variable is not given because there is no chance to call `handle_signal(timeout)` in the forked child.

Before:

* TERM: Set `@shutdown = true`, but actually affects nothing
* INT: Set `@shutdown = true`, but actually affects nothing
* QUIT: Set `@shutdown = true`, but actually affects nothing
* USR1: Do nothing
* USR2: run_hook :before_pause and set `@paused = true`
* CONT: Set `@paused = false` and run_hook :after_pause

After:

* All signals are reverted to DEFAULT

Which do you prefer the thread version or no thread version?

BTW: Rather, I did not get why resque was not calling unregister_signal_handlers for a forked child always. I felt as `Before` behavior was wierd because it disabled TERM or INT signals.